buildspec.yml

```diff
@@ -0,0 +1,123 @@
+version: 0.2
+
+env:
+  variables:
+    FRAMEWORK_VERSION: '1.0.0'
+    CPU_PY_VERSION: '2'
+    CPU_INSTANCE_TYPE: 'ml.c4.xlarge'
+    GPU_PY_VERSION: '3'
+    GPU_INSTANCE_TYPE: 'ml.p2.xlarge'
+    LOCAL_BASE_REPO: 'pytorch-base'
+    ECR_REPO: 'sagemaker-test'
+    GITHUB_REPO: 'sagemaker-pytorch-container'
+    SETUP_FILE: 'setup_cmds.sh'
+    SETUP_CMDS: '#!/bin/bash\npip install --upgrade pip\npip install -U -e .\npip install -U -e .[test]'
+
+
+phases:
+  pre_build:
+    commands:
+      - start-dockerd
+      - |
+        ACCOUNT=$(aws sts get-caller-identity --query 'Account' --output text)
+        PREPROD_IMAGE="$ACCOUNT.dkr.ecr.$AWS_DEFAULT_REGION.amazonaws.com/$ECR_REPO"
+        PR_NUM=$(echo $CODEBUILD_SOURCE_VERSION | grep -o '[0-9]\+')
+        echo "Pull request number: $PR_NUM. No value means this build is not from pull request."
+
+  build:
+    commands:
+      # install
+      - pip3 install -U -e .
+      - pip3 install -U -e .[test]
+
+      # run unit tests
+      - pytest test/unit
+
+      # build cpu base image
+      - |
+        base_dir="docker/$FRAMEWORK_VERSION/base"
+        cpu_base_tag="$FRAMEWORK_VERSION-cpu-py$CPU_PY_VERSION"
+        cpu_dockerfile="Dockerfile.cpu"
+        cd $base_dir
+        docker build -t $LOCAL_BASE_REPO:$cpu_base_tag -f $cpu_dockerfile --build-arg py_version=$CPU_PY_VERSION .
+        cd ../../../
+
+      # build gpu base image
+      - |
+        gpu_base_tag="$FRAMEWORK_VERSION-gpu-py$GPU_PY_VERSION"
+        gpu_dockerfile="Dockerfile.gpu"
+        cd $base_dir
+        docker build -t $LOCAL_BASE_REPO:$gpu_base_tag -f $gpu_dockerfile --build-arg py_version=$GPU_PY_VERSION .
+        cd ../../../
+
+      # create wheel
+      - python3 setup.py bdist_wheel
+
+      # build cpu image
+      - |
+        build_dir="docker/$FRAMEWORK_VERSION/final"
+        build_id="$(echo $CODEBUILD_BUILD_ID | sed -e 's/:/-/g')"
+        CPU_TAG="$FRAMEWORK_VERSION-cpu-py$CPU_PY_VERSION-$build_id"
+        docker build -f "$build_dir/$cpu_dockerfile" --build-arg py_version=$CPU_PY_VERSION -t $PREPROD_IMAGE:$CPU_TAG .
+
+      # build gpu image
+      - |
+        GPU_TAG="$FRAMEWORK_VERSION-gpu-py$GPU_PY_VERSION-$build_id"
+        docker build -f "$build_dir/$gpu_dockerfile" --build-arg py_version=$GPU_PY_VERSION -t $PREPROD_IMAGE:$GPU_TAG .
+
+      # push images to ecr
+      - |
+        $(aws ecr get-login --registry-ids $ACCOUNT --no-include-email --region $AWS_DEFAULT_REGION)
+        docker push $PREPROD_IMAGE:$CPU_TAG
+        docker push $PREPROD_IMAGE:$GPU_TAG
+
+      # launch remote gpu instance
```

Comment:
same suggestion as with https://github.com/aws/sagemaker-chainer-container/pull/80 - maybe move this to after the CPU tests